ngx_http_apicast_module.c

```diff
@@ -114,6 +116,32 @@ int ngx_http_apicast_ffi_set_proxy_cert_key(
 
   ctx->proxy_client_cert = x509;
   ctx->proxy_client_cert_key = cert_key;
+
+  if ( number_of_certs > 1 ) {
+    STACK_OF(x509) *chain = sk_X509_new_null();
+    if (chain == NULL) {
+        err = "sk_X509_new_null() failed";
+        goto failed;
+    }
+
+    X509 *x509_chain = NULL;
+    for (int i = 1; i < number_of_certs; i++) {
+      x509_chain = sk_X509_value(cert_chain, i);
+      if (x509 == NULL) {
+        err = "sk_X509_value() failed on chain certificate";
+        goto failed;
```

Comment:
Note that in this case `chain` is not freed, which perhaps only would have the effect of a memory leak. I'd recommend that you create a new label for `chain_failed` and jump there freeing `chain`, then jump there as well below after freeing `x509_chain`.